src/taipy/core/scenario/_scenario_manager.py

```diff
@@ -280,6 +268,27 @@ def _compare(cls, *scenarios: Scenario, data_node_config_id: Optional[str] = Non
     def __get_config(scenario: Scenario):
         return Config.scenarios.get(scenario.config_id, None)
 
+    @classmethod
+    def _is_deletable(cls, scenario: Union[Scenario, ScenarioId]):
+        if isinstance(scenario, str):
+            scenario = cls._get(scenario)
+        if scenario.is_primary:
+            if len(cls._get_all_by_cycle(scenario.cycle)) > 1:
+                return False
+        return True
+
+    @classmethod
+    def _delete(cls, scenario_id: ScenarioId):
+        scenario = cls._get(scenario_id)
+        if scenario.is_primary:
+            if len(cls._get_all_by_cycle(scenario.cycle)) > 1:
```

Comment:
Maybe we can refactor this part, something like if not _is_deletable then raise error.
And the `_hard_delete()` method as well.
What do you think?

Reply:
Yes, good idea.